### PR TITLE
merge dev into block Merge

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview

merge dev into block Merge

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Review by nzpr — APPROVED** [review]
LGTM

File: shared/src/main/scala/coop/rchain/store/KeyValueTypedStoreSyntax.scala [hunk 1]
```diff
@@ -17,7 +18,18 @@ final class KeyValueTypedStoreOps[F[_], K, V](
...
+      kvIfAbsent = kvPairs zip ifAbsent
+      kvAbsent   = kvIfAbsent.filterNot(_._2)
+      _          <- store.put(kvAbsent.map(_._1))
+    } yield ()
+
   def delete(key: K)(implicit f: Functor[F]): F[Boolean] = store.delete(Seq(key)).map(_ == 1)
 
   def contains(key: K)(implicit f: Functor[F]): F[Boolean] = store.contains(Seq(key)).map(_.head)
+
+  def getOrElse(key: K, elseValue: V)(implicit f: Functor[F]): F[V] =
```
Comment (tgrospic) [inline shared/src/main/scala/coop/rchain/store/KeyValueTypedStoreSyntax.scala:33]: Please create an issue to fix this. `elseValue` is executed even if element is found, it must be converted to thunk parameter.

**Review by tgrospic — APPROVED** [review]
LGTM

Please leave the commit message with detected conflicts when doing merge. Example is last commit on on this PR. It's easier to track what is more important to look into.
https://github.com/rchain/rchain/pull/3152/commits

I added one comment that can be fixed in separate PR.

Good work!

File: shared/src/main/scala/coop/rchain/store/KeyValueTypedStoreSyntax.scala (hunk 1 above)
Comment (zsluedem) [reply shared/src/main/scala/coop/rchain/store/KeyValueTypedStoreSyntax.scala:33]: Do you mean something like ` elseValue: =>V`? I don't actually understand what you mean.

**Comment by zsluedem** [comment]
bors merge

File: shared/src/main/scala/coop/rchain/store/KeyValueTypedStoreSyntax.scala (hunk 1 above)
Comment (tgrospic) [reply shared/src/main/scala/coop/rchain/store/KeyValueTypedStoreSyntax.scala:33]: Yes, exactly. It's like creating a function which is not evaluated if not used.

Comment (tgrospic) [reply shared/src/main/scala/coop/rchain/store/KeyValueTypedStoreSyntax.scala:33]: In the same way how Option is implemented for `getOrElse`.
https://github.com/scala/scala/blob/11af53f/src/library/scala/Option.scala#L188

**Pull request merged**